### PR TITLE
Issue544 retail profiles

### LINKED ISSUE #544: Integrating Advanced Building Simulation
During the B13 Project we want to integrate some new features into TEASER. This issue should be used as a roadmap for different implentations. Further Information will be added.

## What is the problem?

- ...

## Why do we want to solve it?

- ...

## How do we want to solve it?

- ...

### PR DESCRIPTION
**Pull request by DaJansenGit**
This PR brings some options to make manipulate the existing profiles to create a bit more dynamic. Main functions are:
* timeset and scaling factor to reduce (or increase) the profiles on weekend for persons, lighting and machines
* opening-, closing times to cut/extend existing profiles for persons, lighting and machines
* heating/cooling set back temperatures and time schedules to allow easy night set back

This should also close #544 as air curtain already exists and only retail profiles were missing.

### CODE REVIEW

**Review by MartinRaetz — APPROVED** [review]

File: teaser/logic/buildingobjects/useconditions.py [hunk 1]
```diff
@@ -480,9 +624,126 @@ def lighting_profile(self):
...
+        if self.is_periodic(value):
+            self._lighting_profile = value
+        else:
+            raise ValueError(
+                f"lighting profile should be periodic (24h, 168h pr 8760h), "
+                "but length is {len(value)}"
+            )
+
+    @property
+    def schedules(self):
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply teaser/logic/buildingobjects/useconditions.py:637]: I see your point here. To still hold this easy to use and backwards compatible, I left the original schedules attribute as a property and moved the adjustments to a function called `calc_adj_schedules()`. This way we don't need to adjust the exports as schedules are always calculated when called, but to make changes to the schedules based on the previous set adjustment attributes like `adjusted_opening_times` one needs to call the  `calc_adj_schedules()` function afterwards. This should be a good combination from explicit and implicit functionality.

File: tests/test_useconditions.py [hunk 2]
```diff
@@ -142,3 +144,60 @@ def test_ahu_threshold_false(self):
         use_cond.with_ahu = False
         with pytest.raises(Exception):
             use_cond.with_ideal_thresholds = True
+
+    def test_profile_adjust_opening_times(self):
+        prj.set_default()
+        helptest.building_test2(prj)
+        use_cond = prj.buildings[-1].thermal_zones[-1].use_conditions
+        use_cond.adjusted_opening_times = [10, 15]
+        profile_before = use_cond.machines_profile
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply tests/test_useconditions.py:153]: there is not overwriting done in line 152, only getting the values. But anyway I agree that we can switch line 152 and 153 to improve readability. Did this for the other tests as well.

**Comment by DaJansenGit** [comment]
Sorry that this took so long, I finally made the requested changes. @MichaMans are you still available for a small review? :) Otherwise, I will ask someone else. Thanks in advance.

**Comment by DaJansenGit** [comment]
@Dawac Could you have a look on this?

**Review by Dawac — APPROVED** [review]
Nice work, seems to work as intended. 
Maybe we can add a method to smooth shortend machine, lightning etc. profiles in the future. As is, the profile is just shifted to the adjusted opening hours and then cut of at the adjusted closing hour.

**Pull request merged**